Workflow/Sources/ApplyContext.swift

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// Internal utility protocol so that `WorkflowTesting` can provide an alternate implementation
+protocol ApplyContextType<WorkflowType> {
+    associatedtype WorkflowType: Workflow
+
+    subscript<Value>(
+        workflowValue keyPath: KeyPath<WorkflowType, Value>
+    ) -> Value { get }
+}
+
+/// Runtime context passed as a parameter to `WorkflowAction`'s `apply()` method
+/// that provides an integration point with the runtime that can be used to read property values
+/// off of the current `Workflow` instance.
+public struct ApplyContext<WorkflowType: Workflow> {
+    let wrappedContext: any ApplyContextType<WorkflowType>
+
+    init<Impl: ApplyContextType>(_ context: Impl)
+        where Impl.WorkflowType == WorkflowType
+    {
+        self.wrappedContext = context
+    }
+
+    public subscript<Value>(
```

Comment:
This is missing docs, but I think it would actually be more effective to document usage on the `ApplyContext` type, with a small example of how to read workflow values.

Reply:
added some more context & a usage example to the doc comment on the containing type